junit4-ant/src/main/java/com/carrotsearch/ant/tasks/junit4/runlisteners/UserDefinedRunListeners.java

```diff
@@ -0,0 +1,26 @@
+package com.carrotsearch.ant.tasks.junit4.runlisteners;
+
+import org.junit.runner.notification.RunListener;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class UserDefinedRunListeners {
+
+    public static ArrayList<RunListener> generateInstances() throws Exception {
+        ArrayList<RunListener> instances = new ArrayList<>();
+
+        String userDefinedRunListeners = System.getProperty("userDefinedRunListeners");
```

Comment:
I somehow don't like this being passed via system property. This isn't really needed: the listener list can be passed to SlaveMain just like other flags (see the OPTION_* parsing in the main method)?

I'd also just move the whole instantiation to slavemain instead of creating a (public) method in a separate (public) class.

Reply:
I wasn't sure the best way to get these values to SlaveMain.  I will look at passing via flags like you had suggested and will update the PR

Reply:
This is just forked as a command-line process, so command-line args feel more natural to me (not to mention they don't pollute the global property pool). The only reason other tests.* options (timeouts, seeds) are passed as system properties is that it allows they override in IDEs (where SlaveMain isn't even present, just the runner itself).

Reply:
Got it.  I will make the changes and update the PR